Doc/c-api/init.rst

```diff
@@ -307,6 +314,47 @@ Initializing and finalizing the interpreter
    disregards the return value.
 
 
+.. c:function:: int Py_Main(int argc, wchar_t **argv)
+
+   The main program for the standard interpreter, encapsulating a full
+   :c:func:`Py_Initialize`/:c:func:`Py_Finalize` cycle, as well as additional
+   behaviour to implement reading configurations settings from the environment
+   and command line, and then executing ``__main__`` in accordance with
+   :ref:`using-on-cmdline`.
+
+   This is made available for programs which wish to support the full CPython
+   command line interface, rather than just embedding a Python runtime in a
+   larger application.
+
+   The *argc* and *argv* parameters should be prepared exactly as those which
+   are passed to a C program's :c:func:`main` function (converted to ``wchar_t``
+   with :c:func:`Py_DecodeLocale`).  It is important to note that the argument
+   list may be modified (but the contents of the strings pointed to by the
+   argument list are not).
+
+   The return value will be ``0`` if the interpreter exits normally (i.e.,
+   without an exception), ``1`` if the interpreter exits due to an exception,
+   or ``2`` if the argument list does not represent a valid Python command
+   line.
+
+   Note that if an otherwise unhandled :exc:`SystemExit` is raised, this
+   function will not return ``1``, but exit the process, as long as
+   ``Py_InspectFlag`` is not set. If ``Py_InspectFlag`` is set, execution will
+   drop into interactive Python prompt, at which point a second otherwise
+   unhandled :exc:`SystemExit` will still exit the process, while any other
+   means of exiting will set the return value as described above.
+
+   In normal usage, an embedding application will call this function
+   *instead* of calling :c:func:`Py_Initialize` or :c:func:`Py_InitializeEx`
+   directly, and all settings will be applied as described elsewhere in this
+   documentation. If this function is instead called *after* a preceding
+   ``Py_Initialize`` or ``Py_InitializeEx`` call, then exactly which
+   environmental and command line configuration settings will be updated is
+   version dependent (as it depends on which settings correctly support
+   being modified after they have already been set once in the first call to
+   ``Py_Initialize`` or ``Py_InitializeEx``).
```

Comment:
Can you please try to document Py_RunMain()? It may help to explain how Py_Main() is implemented.

Py_Main() pseudo-code (without error handling) looks like:

```
PyConfig config;
PyConfig_InitPythonConfig(&config);
PyConfig_SetArgv(&config, argc, argv);
Py_InitializeFromConfig(&config);
PyConfig_Clear(&config);

Py_RunMain();
```

Reply:
Done (I also moved the docs for `PyConfig_InitPythonConfig` and `Py_RunMain` into this page)